packages/snap-toolbox/src/DomTargeter/DomTargeter.ts

```diff
@@ -27,19 +30,63 @@ export class DomTargeter {
 
 		this.retarget();
 
-		if (/complete|loaded/.test(this.document.readyState)) {
-			// DOMContent has loaded - unhide targets
-			this.targets.forEach((target) => target.hideTarget && this.unhideTarget(target.selector));
+		//find out if we are auto retargeting
+		this.targets.forEach((target) => {
+			if (target.autoRetarget) {
+				this.autoRetarget = true;
+			}
+		});
+
+		if (this.autoRetarget) {
+			this.targets.forEach((target) => {
+				const checker = () => {
+					//lets not just keep trying forever
+					if (this.timeoutTime < 1000) {
```

Comment:
Should all targets be using the same timeoutTime? If you had like 5 targets, wouldn't they all bump this over 1000 after the first check?

Perhaps each target should have its own timeout timer - wouldn't need to be a property on the target, but could maybe just be a scoped variable?